lib/elastic_record/percolator_model.rb

```diff
@@ -0,0 +1,51 @@
+module ElasticRecord
+  module PercolatorModel
+    def self.included(base)
+      base.class_eval do
+        class_attribute :target_model
+
+        include Model
+        extend ClassMethods
+      end
+    end
+
+    module ClassMethods
+      def elastic_index
+        @elastic_index ||=
+          begin
+            index = ElasticRecord::Index.new([self, target_model])
+            index.partial_updates = false
+            index.settings = target_model.elastic_index.settings.slice('analysis')
```

Comment:
This is kind of gross.  What do you think about moving analyzer configuration into the doctype, and then having the index build its `settings.analysis` from the registered doctypes, like I'm doing with mapping now?